src/components/transactionsV2/filters/amountFieldGroup.js

```diff
@@ -61,44 +69,72 @@ class AmountFieldGroup extends React.Component {
+    const fieldsObj = Object.keys(filters).reduce((acc, filter) =>
       ({ ...acc, [filter]: { value: filters[filter] } }), {});
 
-    this.validateAmountField({
+    const fields = {
       ...fieldsObj,
-      [target.name]: { value },
-    });
+      [target.name]: { value, loading: true },
+    };
+
+    this.setState({ fields });
+
+    clearTimeout(this.timeout);
+    this.timeout = setTimeout(() => {
+      this.validateAmountField(fields);
+    }, 300);
+
+    this.props.updateCustomFilters(fields);
   }
 
-  render() {
-    const { filters, handleKeyPress, t } = this.props;
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  generateField(data) {
+    const { filters, handleKeyPress } = this.props;
     const { fields } = this.state;
 
+    return (
+      <label className={styles.fieldHolder}>
+        <InputV2
+          autoComplete={'off'}
+          onChange={this.handleFieldChange}
+          name={data.name}
+          value={filters[data.name]}
+          placeholder={data.placeholder}
+          onKeyDown={handleKeyPress}
+          className={`${styles.input} ${fields[data.name].error ? 'error' : ''} ${data.name}Input`}
+        />
+        <SpinnerV2
+          className={`${styles.status} ${fields[data.name].loading && filters[data.name] ? styles.show : ''}`}
+        />
+        <img
+          className={`${styles.status} ${!fields[data.name].loading && filters[data.name] ? styles.show : ''}`}
+          src={ fields[data.name].error ? svg.alert_icon : svg.ok_icon}
+        />
+      </label>
+    );
+  }
+
+  render() {
+    const { t } = this.props;
+
     return (
       <div className={styles.fieldGroup}>
         <span className={styles.fieldLabel}>{t('Amount')}</span>
         <div className={styles.fieldRow}>
-          <InputV2
-            autoComplete={'off'}
-            onChange={this.handleFieldChange}
-            name='amountFrom'
-            value={filters.amountFrom}
-            placeholder={t('Min')}
-            onKeyDown={handleKeyPress}
-            className={`${styles.input} ${fields.amountFrom.error ? 'error' : ''} amountFromInput`} />
-          <span>-</span>
-          <InputV2
-            autoComplete={'off'}
-            onChange={this.handleFieldChange}
-            name='amountTo'
-            value={filters.amountTo}
-            placeholder={t('Max')}
-            onKeyDown={handleKeyPress}
-            className={`${styles.input} ${fields.amountTo.error ? 'error' : ''} amountToInput`} />
+          { this.generateField({ name: 'amountFrom', placeholder: t('Min') }) }
+          <span className={styles.separator} />
+          { this.generateField({ name: 'amountTo', placeholder: t('Max') }) }
         </div>
-        <span className={`${styles.feedback} ${this.state.feedback ? styles.show : ''}`}>
-          {this.state.feedback}
-        </span>
+        <Feedback
+          className={styles.feedback}
+          show={!!this.state.feedback}
```

Comment:
I think in here you can use error value too, as it is a bool

Reply:
On this case it would be needed to test both fields error value, this way we only need to check if we have a feedback message for any of the fields.